tools/python/util/convert_onnx_models_to_ort.py

```diff
@@ -153,31 +163,16 @@ def parse_args():
-                             'operators. Note that this will limit the optimizations possible on nodes that the '
-                             'NNAPI execution provider takes, in order to preserve those nodes in the ORT format '
-                             'model.')
-
-    parser.add_argument('--use_coreml', action='store_true',
-                        help='Enable the CoreML Execution Provider when creating models and determining required '
-                             'operators. Note that this will limit the optimizations possible on nodes that the '
-                             'CoreML execution provider takes, in order to preserve those nodes in the ORT format '
-                             'model.')
-
-    parser.add_argument('--optimization_level', default=['basic', 'all'], nargs='+',
-                        choices=['disable', 'basic', 'extended', 'all'],
-                        help="Level to optimize ONNX model with, prior to converting to ORT format model. "
-                             "These map to the onnxruntime.GraphOptimizationLevel values. "
-                             "If the level is 'all' the NCHWc transformer is manually disabled as it contains device "
-                             "specific logic, so the ORT format model must be generated on the device it will run on. "
-                             "Additionally, the NCHWc optimizations are not applicable to ARM devices. "
-                             "Multiple values can be provided. A model produced with 'all' is optimal for usage with "
-                             "just the CPU Execution Provider. A model produced with 'basic' is required for usage "
-                             "with the NNAPI or CoreML Execution Providers. "
-                             "The filename for the ORT format model will contain the optimization level that was used "
-                             "to create it."
-                        )
+    parser.add_argument('--optimization_style',
+                        default=OptimizationStyle.Default.name, choices=[e.name for e in OptimizationStyle],
+                        help="Style of optimization to perform on the ORT format model. "
+                             f"'{OptimizationStyle.Default.name}': Run optimizations directly before saving the ORT "
+                             "format model. "
+                             f"'{OptimizationStyle.Runtime.name}': Run basic optimizations directly and save certain "
+                             "other optimizations to be applied at runtime if possible. This is useful when using a "
+                             "compiling EP like NNAPI or CoreML that may run an unknown (at model conversion time) "
+                             "number of nodes. The saved optimizations can further optimize nodes not assigned to the "
+                             "compiling EP at runtime.")
 
```

Comment:
Maybe call out that pre-optimizing means any platform specific stuff is baked in.

I'd also say that runtime is required if you are using NNAPI or CoreML. If we create things like FusedConv nodes that will prevent NNAPI/CoreML running with the model. 

Actually we could maybe be simpler. Use runtime if you want to use NNAPI/CoreML. Use default in all other cases.